Source/Libraries/GSF.Core/Data/Model/SQLSearchFilter.cs

```diff
@@ -0,0 +1,176 @@
+        /// <exception cref="NotSupportedException">Attempted to assign a type that is not supported.</exception>
+        public string Type
+        {
+            get => m_type;
+            set
+            {
+                if (s_validTypes.Contains(value))
+                    m_type = value;
+                else if (value == "query") // This is needed for legacy support so we do not break anything it can be removed in the future.
+                    m_type = value;
+                else
+                    throw new NotSupportedException($"{value} is not a valid type");
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a flag indicating whether this is a pivot column
+        /// in which case the <see cref="SearchableAttribute"/> should be used.
+        /// </summary>
+        public bool IsPivotColumn { get; set; }
+
+        /// <summary>
+        /// Generates the conditional statement to be used in the WHERE clause of the SQL statement.
+        /// </summary>
+        /// <param name="parameters">A list of the parameters to be substituted into the full query.</param>
+        /// <returns>The conditional statement with format string syntax for parameter substitution."</returns>
+        public string GenerateConditional(List<object> parameters)
+        {
+            string escape = "";
+            if (Operator == "LIKE" || Operator == "NOT LIKE")
+                escape = "ESCAPE '$'";
+
+            string query = "";
+            string searchText = SearchText;
+
+            // For legacy support. In the future, remove the query type.
+            if (Type == "query")
+                query = $"{(IsPivotColumn ? "AFV_" : "") + FieldName} {Operator} ";
+            else
+                query = $"[{(IsPivotColumn ? "AFV_" : "") + FieldName.RemoveCharacter('[').RemoveCharacter(']')}] {Operator} ";
+
+            /* This requires switch to the new search format
+                if (Operator == "IN" || Operator == "NOT IN")
+                {
+                    IEnumerable<string> valueList = searchText
+                        .Replace("(", "")
+                        .Replace(")", "")
+                        .Split(',');
+
+                    searchText = $"({string.Join(",", valueList.Select((s, i) => $"{{{i + parameters.Count}}}"))})";
+                    parameters.AddRange(valueList);
+                }
+            */
+
+            /* We update to remove the following once we move to new SQL injection logic,
+             * but it is needed in here for now since I had to leave the logic out
+             * of the above to avoid legacy issues
+             */
+
+            if (Type != "query" && (Operator == "IN" || Operator == "NOT IN"))
+            {
+                IEnumerable<string> valueList = searchText
+                    .Replace("(", "")
+                    .Replace(")", "")
+                    .Split(',')
+                    .Select(t => $"'{t}'");
```

Comment:
So the syntax for IN/NOT IN is going to be something like `AN,BN,CN`? Does the user have to enter the commas? Should we trim each element in the list? What about values that contain commas or parens; do we not care?

Reply:
No the Commas get added by the UI. These are primary for the Filters that show Checkboxes. We need this for legacy reasons. This is currently working and we want to avoid breaking anything for the time being so trimming or considering values with commas may cause issues.